idaes/power_generation/unit_models/boiler_fireside.py

```diff
@@ -176,6 +177,12 @@ class BoilerFiresideData(UnitModelBlockData):
 
     def build(self):
         super(BoilerFiresideData, self).build()
+        # Insert user provided custom surrogate model function
+        # self.config.surrogate_function(self)
+        if self.config.surrogate_dictionary is None:
+            raise ConfigurationError('user needs to provide '
```

Comment:
Error messages really should start with a capital, and it is probably helpful to tell the user where the error occurred. I.e.

`raise ConfiguraitonError('{} Some message here'.format(self.name))`

This way, the user knows which model they need to fix.